doc/man3/SSL_CTX_set_tlsext_servername_callback.pod

```diff
@@ -39,16 +41,25 @@ SSL_get_servername_type() returns the servername type or -1 if no servername
 is present. Currently the only supported type (defined in RFC3546) is
 B<TLSEXT_NAMETYPE_host_name>.
 
+SSL_set_tlsext_host_name() sets the server name indication extension typically
+in ClientHello with value B<name>. The type of server name indication extension
+is set to B<TLSEXT_NAMETYPE_host_name> (defined in RFC3546).
+
 =head1 NOTES
 
 Several callbacks are executed during ClientHello processing, including
 the early, ALPN, and servername callbacks.  The early callback is executed
 first, then the servername callback, followed by the ALPN callback.
 
+The SSL_set_tlsext_host_name() function should not be called from a server
+context (where L<SSL_is_server(3)> returns ture). If called, it just simply
+returns 1 and B<name> is ignored.
+
```

Comment:
I would consider mentioning (either here or in a new HISTORY section) that prior to 1.1.1 the call on a server SSL would succeed and had a potential weird interaction with SSL_SESSIONs generated from that SSL.  But it's a sufficiently obscure/strange thing that I don't feel it strictly needs to be documented, and am okay if you want to just leave this part as-is.